### PR TITLE
chore: python: Use the static keyword for static methods in python

### PR DESCRIPTION
**Pull request by akuhlens**
Simple change to the python parser to lower `@staticmethod` and `@staticmethod()` to the static keyword. This is used to eliminate some python specific details in this [semgrep-pro pr](https://github.com/semgrep/semgrep-proprietary/pull/1298).

# Test Plan
Adds a test to check the behavior of matching. CI should continue to work with the exception of semgrep pro, which is changed to work in the PR above.

### CODE REVIEW

**Review by nmote — APPROVED** [review]
Makes sense. `staticmethod` is a Python builtin, right?

**Comment by akuhlens** [comment]
> `staticmethod` is a Python builtin, right?

This is correct.

File: tests/patterns/python/static_method.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    def f():
+        pass
+
+    #MATCH:
+    @staticmethod()
+    def f():
+        pass
+
+    #OK:
+    @staticmethod("A ERROR")
```
Comment (IagoAbal) [inline tests/patterns/python/static_method.py:19]: Perhaps make clear in the comment that this is _not_ a static method.

Comment (IagoAbal) [reply tests/patterns/python/static_method.py:19]: Maybe not even valid Python? 🤷‍♂️

Comment (akuhlens) [reply tests/patterns/python/static_method.py:19]: Not valid python, just documenting the edge cases, but done.
